analyzers/tests/SonarAnalyzer.Test/TestCases/ArrayPassedAsParams.cs

```diff
@@ -82,31 +82,29 @@ public class Repro6894
 
     public void CallMethod()
     {
-        Method(new String[] { "1", "2" }); // Noncompliant, TP. Elements in args: ["1", "2"]
-                                           // The argument given for a parameter array can be a single expression that is implicitly convertible (§10.2) to the parameter array type.
-                                           // In this case, the parameter array acts precisely like a value parameter.
-                                           // see: https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/classes#14625-parameter-arrays
-        Method(new object[] { new int[] { 1, 2} }); // Noncompliant. TP Elements in args: [System.Int32[]]
-        Method(new int[] { 1, 2, 3, }); // Noncompliant, FP. Elements in args: [System.Int32[]]
-        Method(new String[] { "1", "2" }, new String[] { "1", "2"}); // Noncompliant, FP. Elements in args: [System.String[], System.String[]]
-        //                                ^^^^^^^^^^^^^^^^^^^^^^^^
-        Method(new String[] { "1", "2"}, new int[] { 1, 2}); // Noncompliant, FP. Elements in args: pSystem.String[], System.Int32[]]
-        //                               ^^^^^^^^^^^^^^^^^
-        MethodArray(new String[] { "1", "2" }, new String[] { "1", "2" }); // Noncompliant, FP. Elements in args: [System.String[], System.String[]]
-        MethodArray(new int[] { 1, 2 }, new int[] { 1, 2 }); // Noncompliant, FP. Elements in args: [System.Int32[], System.Int32[]]
-
-        MethodJaggedArray(new int[] { 1, 2 }); // Compliant: jagged array [System.Object[]]
+        Method(new String[] { "1", "2" });                                  // FN. Elements in args: ["1", "2"]
```

Comment:
As discussed offline, the change introduces FNs for all reference types, and not just for `String`.
This has a big impact on the rule, since, when `params object[]` is used in the declaration, it's more likely because a list of reference types (such as `String`), are going to be passed, rather than primitives (which also have implicit promotion rules) and structures (which don't have inheritance).
So, for the cases of primitives, it would be more likely in the vast majority of cases, to specify the concrete numeric type (e.g. `params int[]`), rather than `params object[]`. The same applies to structures: given the lack of inheritance, you probably are not going to see `params object[]` as much as `params ConcreteStruct[]`.